packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItemStyles.ts

```diff
@@ -65,28 +122,59 @@ const useSpreadStyles = makeStyles({
  * Apply styling to the AvatarGroupItem slots based on the state
  */
 export const useAvatarGroupItemStyles_unstable = (state: AvatarGroupItemState): AvatarGroupItemState => {
+  const avatarCount = useContextSelector(AvatarGroupContext, ctx => ctx.avatarCount);
```

Comment:
You're reading other context values in `useAvatarGroupItem` and deriving `state` from them, my feeling is that these context values should also be read there and attached to `state`. That change would make `useAvatarGroupItemStyles` solely a function of its argument which generally makes a function easier to test, debug and re-use.